guide/phone_store/phone_store_service.bal

```diff
@@ -61,171 +52,177 @@ import ballerina/log;
         initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
         providerUrl: "tcp://localhost:61616"
     });
-// Initialize a JMS session on top of the created connection
+
+// Initialize a JMS session on top of the created connection.
 jms:Session orderQueueJmsSessionSend = new(orderQueueJmsConnectionSend, {
         acknowledgementMode: "AUTO_ACKNOWLEDGE"
     });
-// Initialize a queue sender using the created session
-endpoint jms:QueueSender jmsProducerOrderQueue {
-    session: orderQueueJmsSessionSend,
-    queueName: "OrderQueue"
-};
-// Service endpoint
-endpoint http:Listener listener {
-    port: 9090
-};
 
-//backendreq used to obtain details of order queue
-public http:Request backendreq;
+// Initialize a queue sender using the created session.
+jms:QueueSender jmsProducerOrderQueue = new(orderQueueJmsSessionSend, queueName = "OrderQueue");
+
+// Service endpoint.
+listener http:Listener httpListener = new(9090);
+
+http:Request backendreq = new;
 
-// phone store service, which allows users to order phones online for delivery
-@http:ServiceConfig { basePath: "/phonestore" }
-service<http:Service> phoneStoreService bind listener {
-    // Resource that allows users to place an order for a phone
+// Phone store service, which allows users to order phones online for delivery.
+@http:ServiceConfig {
+    basePath: "/phonestore"
+}
+service phoneStoreService on httpListener {
+    // Resource that allows users to place an order for a phone.
     @http:ResourceConfig {
         methods: ["POST"],
         consumes: ["application/json"],
         produces: ["application/json"]
     }
-    placeOrder(endpoint caller, http:Request request) {
-
+    resource function placeOrder(http:Caller caller, http:Request request) {
         backendreq = untaint request;
-        http:Response response;
-        PhoneOrder newOrder;
-        json reqPayload;
-
-        // Try parsing the JSON payload from the request
-        match request.getJsonPayload() {
-            // Valid JSON payload
-            json payload => reqPayload = payload;
-            // NOT a valid JSON payload
-            any => {
-                response.statusCode = 400;
-                response.setJsonPayload({ "Message": "Invalid payload - Not a valid JSON payload" });
-                _ = caller->respond(response);
-                done;
-            }
+        http:Response response = new;
+        PhoneOrder newOrder = {};
+        json requestPayload = {};
+
+        var payload = request.getJsonPayload();
+        // Try parsing the JSON payload from the request.
+        if (payload is json) {
+            // Valid JSON payload.
+            requestPayload = payload;
+        } else {
+            // NOT a valid JSON payload.
+            response.statusCode = 400;
+            response.setJsonPayload({ "Message": "Invalid payload - Not a valid JSON payload" });
+            checkpanic caller->respond(response);
```

Comment:
Any reason for having removed the `done` here?

Reply:
I added `return`